### PR TITLE
Remove some redundant edges from tutorial DAG

### PR DESCRIPTION
**Pull request by krishnakumarg1984**
This fixes a few cases that were probably missed out in #9373

### CODE REVIEW

File: examples/step-54/doc/builds-on [hunk 1]
```diff
@@ -1 +1 @@
-step-49 step-53
+step-53
```
Comment (tjhei) [inline examples/step-54/doc/builds-on]: this is in #9373

File: examples/step-44/doc/builds-on [hunk 2]
```diff
@@ -1 +1 @@
-step-8 step-18
+step-18
```
Comment (tjhei) [inline examples/step-44/doc/builds-on]: this is correct

File: examples/step-12/doc/builds-on [hunk 3]
```diff
@@ -1 +1 @@
-step-7 step-39
+step-39
```
Comment (tjhei) [inline examples/step-12/doc/builds-on]: I am not sure about this. I would probably remove 39 instead as the prerequisite for 12 instead. I have plans to introduce a new step in between anyways. Can you remove this for now?

File: examples/step-54/doc/builds-on (hunk 1 above)
Comment (krishnakumarg1984) [reply examples/step-54/doc/builds-on]: Oops. Didn't see

File: examples/step-44/doc/builds-on (hunk 2 above)
Comment (krishnakumarg1984) [reply examples/step-44/doc/builds-on]: Ok. So, we will retain this.

File: examples/step-12/doc/builds-on (hunk 3 above)
Comment (krishnakumarg1984) [reply examples/step-12/doc/builds-on]: ok!

**Comment by bangerth** [comment]
Same here: Please squash/fixup.

**Comment by krishnakumarg1984** [comment]
Done!

**Pull request merged**